Week-4/assignment4_2.js

```diff
@@ -0,0 +1,24 @@
+var Person = function(){}
+
+Person.prototype.initialize = function (name, age) {
+    this.name = name;
+    this.age = age;
+}
+
+
+function Teacher() {
+}
+
+Teacher.prototype.teach = function (subject, name,age) {
+    const t = Person.call(name, age);
+    
+    console.log(subject);
+
+}
+
+Object.setPrototypeOf(Teacher.prototype, Person.prototype)
```

Comment:
through this you are overriding the teacher prototype , ideally this should come abpve line 12